### PR TITLE
Fix some issues with C templating.

### PR DESCRIPTION
**Pull request by daxida**
Tested it against this: [scrabble](https://www.codingame.com/contribute/view/782593a90df79da5f0d30900aec1cdbdbe36a) and it works fine.

I changed `tmp.*` to `tmp*` in the `.gitignore` to also ignore the binary generated by C, and a quick command to the `justfile` for testing.

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -71,6 +71,12 @@ launch-py:
     cargo run --quiet -- show
     ls *.py | entr -p cargo run --quiet -- run --command "python3 tmp.py"
 
+launch-c:
+    code tmp.c
```
Comment (ellnix) [inline justfile]: Maybe we should use $EDITOR or some other paradigm? Hardcoding VS Code seems short sighted

File: src/stub/parser.rs [hunk 2]
```diff
@@ -54,7 +54,7 @@ impl<'a, I: Iterator<Item = &'a str>> Parser<I> {
                     Some("\n") | None => break,
                     Some(str) => format!("\n{}", str),
                 },
-                join if join.contains("join(") => return self.parse_write_join(join),
+                join if join.starts_with("join(") => return self.parse_write_join(join),
```
Comment (ellnix) [inline src/stub/parser.rs:57]: :+1:

File: src/stub/parser.rs [hunk 3]
```diff
@@ -270,11 +270,19 @@ impl<'a, I: Iterator<Item = &'a str>> Parser<I> {
...
             text_block.push(next_token);
         }
 
-        text_block.join(" ")
+        // Hacky - The replace is due to the "replace hacks" at generator
+        // so that ["you", "\ndown"] -> you\ndown ...
+        // while trimming spaces both sides of each line
+        text_block
+            .join(" ")
+            .replace(" \n", "\n")
```
Comment (ellnix) [inline src/stub/parser.rs:281]: Yeah I'll look into fixing this, these hacks are getting too much

**Review by ellnix — APPROVED** [review]

File: justfile (hunk 1 above)
Comment (daxida) [reply justfile]: That seems simple to add. Will do.

File: justfile [hunk 4]
```diff
@@ -53,26 +53,30 @@ check-not-matching:
 ###################
 # HERE BE DRAGONS #
 ###################
+
+# Change this to your favorite text-editor
+editor := "code"
```
Comment (ellnix) [inline justfile:58]: :+1:

**Review by ellnix — APPROVED** [review]

File: config/stub_templates/rust/stub_config.toml [hunk 5]
```diff
@@ -13,5 +13,5 @@ keywords = [
 Int = "i32"
 Long = "i64"
 Float = "f64"
-Bool = "to_bool"
+Bool = "i32"
```
Comment (ellnix) [inline config/stub_templates/rust/stub_config.toml:16]: Not `bool`?

File: config/stub_templates/c/stub_config.toml [hunk 6]
```diff
@@ -9,7 +9,7 @@ keywords = [
 [type_tokens]
 Int = "int"
 Long = "long long"
-Float = "TODO"
-Bool = "TODO"
+Float = "float"
+Bool = "int"
```
Comment (ellnix) [inline config/stub_templates/c/stub_config.toml:13]: Same here as in the rust config

Comment (daxida) [reply config/stub_templates/c/stub_config.toml:13]: That's how it's parsed inCG for C / rust, as an int.

To this `read n:bool` you get:

image
image

It is a non issue, booleans are not used and this is (arguably) just better than a TODO.

**Pull request merged**